### PR TITLE
Config changes

### PR DESCRIPTION
**Pull request by cfpwastaken**
- serverName is now host
- Added example config.json
- getConfig() now accepts a default value
- SMTPClient now uses the host from the config
- SMTPServer can now be configured to use a custom SMTP Header

### CODE REVIEW

File: config.sample.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "host": "example.com",
+  "db_dialect": "mariadb",
+  "db_username": "root",
+  "db_password": "secret",
+  "db_database": "mailverse",
+  "smtp_header": "example.com SMTP"
```
Comment (j0code) [inline config.sample.json:7]: why not sth. like "%s SMTP" ?
%s is then replaced with host config

File: src/config.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@ import { readFile } from "fs/promises"
 
 export const config = existsSync("config.json") ? JSON.parse(await readFile("config.json", "utf-8")) : {}
 
-export default function getConfig(key: string) {
+export default function getConfig(key: string, defaultValue?: any) {
 	if(config[key]) return config[key]
 	if(process.env[key]) return process.env[key]
+	if(defaultValue) return defaultValue
```
Comment (j0code) [inline src/config.ts:9]:
```suggestion
	if(defaultValue != undefined) return defaultValue
```
"" is falsy

File: src/smtp/SMTPClient.ts [hunk 3]
```diff
@@ -11,7 +12,7 @@ export default class SMTPClient {
 		})
 		sock.on("connect", async () => {
 			console.log("[SMTPClient] Connected to server")
-			sock.write("EHLO localhost\r\n")
+			sock.write("EHLO " + getConfig("host", "localhost") + "\r\n")
```
Comment (j0code) [inline src/smtp/SMTPClient.ts:15]:
```suggestion
			sock.write(`EHLO ${getConfig("host", "localhost")}\r\n`)
```
use template literals :)

File: src/smtp/SMTPServer.ts [hunk 4]
```diff
@@ -20,7 +20,7 @@ export default class SMTPServer {
 
 	connection(sock: net.Socket) {
 		console.log("[SMTP] Client connected")
-		sock.write("220 localhost SMTP Mailverse\r\n")
+		sock.write("220 " + getConfig("smtp_header") + "\r\n")
```
Comment (j0code) [inline src/smtp/SMTPServer.ts:23]:
```suggestion
		sock.write(`220  ${getConfig("smtp_header")}\r\n`)
```
same here

**Review by j0code — CHANGES_REQUESTED** [review]
template literals are richtig and wichtig

File: config.sample.json (hunk 1 above)
Comment (cfpwastaken) [reply config.sample.json:7]: It is supposed to be fully customizable, so not require the hostname. You can do it manually.

**Review by j0code — APPROVED** [review]
fabulous, merge

**Pull request merged**